src/components/FormEditProfile/index.tsx

```diff
@@ -0,0 +1,226 @@
+"use client";
+import { useForm } from "react-hook-form";
+import React, { useState, useEffect } from "react";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { AiFillEye, AiFillEyeInvisible } from "react-icons/ai";
+import * as yup from "yup";
+
+export interface FormEditProfile {
+  username: string;
+  email: string;
+  password: string;
+  confirmPassword: string;
+  newPassword: string;
+}
+
+const schema = yup.object({
+  username: yup.string().required("Nome de usuário é obrigatório"),
+  email: yup.string().email("E-mail inválido").required("E-mail é obrigatório"),
+  password: yup
+    .string()
+    .min(8, "A senha deve ter no mínimo 8 caracteres")
+    .required("Senha é obrigatória"),
+  newPassword: yup.string().required("Senha é obrigatória"),
+  confirmPassword: yup
+    .string()
+    .oneOf([yup.ref("newPassword")], "As senhas devem coincidir")
+    .required("Confirmação de senha é obrigatória"),
+}).required();
+
+export default function FormRegistration() {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    watch,
+  } = useForm<FormEditProfile>({
+    resolver: yupResolver(schema),
+  });
+
+  const [showPassword, setShowPassword] = useState(false);
+  const [showConfirmPassword, setShowConfirmPassword] = useState(false);
+  const [selected, setSelected] = useState<string | null>(null);
+  const [isButtonDisabled, setIsButtonDisabled] = useState(true);
+
+  const togglePasswordVisibility = () => {
+    setShowPassword(!showPassword);
+  };
+
+  const toggleConfirmPasswordVisibility = () => {
+    setShowConfirmPassword(!showConfirmPassword);
+  };
+
+  const handleCheckboxChange = (value: string) => {
+    setSelected(value);
+  };
+
+  const onSubmit = (data: FormEditProfile) => {
+    console.log(data);
+  };
+
+  useEffect(() => {
+    const subscription = watch((value, { name, type }) => {
+      setIsButtonDisabled(false);
```

Comment:
Quando o botão estiver desabilitado, ele deve ter um estilo diferente, que indique ao usuário que está desabilitado (ex.: cor cinza, cursor 'not-allowed', etc.)